src/main/java/com/geoxp/geo/HHCodeHelper.java

```diff
@@ -437,14 +437,14 @@ public static final void stableSplitHHCode(final long hhcode, final int resoluti
    * @return
    */
   public static final long getCenter(long hhcode, int resolution) {
-    // Limit 'hhcode' to 'resolution'
-    if (resolution >= 2 && resolution <= 32) {
-      hhcode = hhcode & Coverage.PREFIX_MASK[(resolution >>> 1) - 1];
-    }
-
-    // Add the 'center bits' for the resolution just above 'resolution'
-    if (resolution >= 2 && resolution < 32) {
-      hhcode = hhcode | Coverage.CENTER_BITS[((resolution >>> 1) - 1) + 1];
+    if (0 == resolution) {
+      // Special case needed because << 64 will result in no shift.
+      hhcode = 0xC000000000000000L;
+    } else if (resolution < 32) {
+      // Limit 'hhcode' to 'resolution'
+      hhcode &= 0xffffffffffffffffL << (64 - resolution * 2);
```

Comment:
Given resolutions are assumed to be even within [2,32] everywhere else I think a detailed description of the semantics given to odd resolutions (and 0) should appear in the javadoc comment of the function, otherwise it will prove difficult to grasp.

Reply:
Although this is true that most functions expects even resolutions, northHHCode and such accepts odd resolutions. I improved the javadoc for `getCenter` to be consistent with those functions.